db/commitlog/commitlog.cc

```diff
@@ -1841,6 +1841,64 @@ future<db::rp_handle> db::commitlog::add_entry(const cf_id_type& id, const commi
     });
 }
 
+future<std::vector<db::rp_handle>> 
+db::commitlog::add_entries(std::vector<commitlog_entry_writer> entry_writers, db::timeout_clock::time_point timeout) {
+    class cl_entries_writer final : public entry_writer {
+        std::vector<commitlog_entry_writer> _writers;
+    public:
+        std::vector<rp_handle> res;
+
+        cl_entries_writer(force_sync sync, std::vector<commitlog_entry_writer> entry_writers)
+            : entry_writer(sync, entry_writers.size()), _writers(std::move(entry_writers))
+        {
+            res.reserve(_writers.size());
+        }
+        const cf_id_type& id(size_t i) const override {
+            return _writers.at(i).schema()->id();
+        }
+        size_t size(segment& seg) override {
+            size_t res = 0;
+            for (auto i = _writers.begin(), e = _writers.end(); i != e; ++i) {
+                auto known = seg.is_schema_version_known(i->schema());
+                if (!known) {
+                    for (auto j = _writers.begin(); j != i; ++j) {
```

Comment:
This can become quadratic, maybe use unordered_set for "known only locally".

Reply:
I opted against it assuming that the number of mutations in would still be relatively few, and the burden of a bunch or allocations in yet another collection was bigger than a few iterations + compares... Maybe I am wrong?

Reply:
The allocations would be the only in the !known case, so once per segment.

I will use only a few mutations in my use, but who knows what happens later?